compiler/sem/semcall.nim

```diff
@@ -429,19 +473,33 @@ proc inferWithMetatype(c: PContext, formal: PType,
       result = copyTree(arg)
       result.typ = formal
 
-proc updateDefaultParams(call: PNode) =
+proc updateDefaultParams(conf: ConfigRef, call: PNode): PNode =
   # In generic procs, the default parameter may be unique for each
   # instantiation (see tlateboundgenericparams).
   # After a call is resolved, we need to re-assign any default value
   # that was used during sigmatch. sigmatch is responsible for marking
   # the default params with `nfDefaultParam` and `instantiateProcType`
   # computes correctly the default values for each instantiation.
   let calleeParams = call[0].sym.typ.n
+  var hasError = false
   for i in 1..<call.len:
     if nfDefaultParam in call[i].flags:
       let def = calleeParams[i].sym.ast
       if nfDefaultRefsParam in def.flags: call.flags.incl nfDefaultRefsParam
-      call[i] = def
+      call[i] =
+        case def.kind
+        of nkEmpty:
+          # ``instantiateProcType`` uses ``nkEmpty`` to signal an incompatible
+          # default expression
+          conf.newError(calleeParams[i],
```

Comment:
```suggestion
          hasError = true
          conf.newError(calleeParams[i],
```

I might have missed it, but I think that's required.

Reply:
Yep, it's required.